EyeSeeTea/user-extended-app
# add d2-logger after saving users

### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/8694wqxd5

### :memo: Implementation

- [x] Settings page for tracker program. 
- [x] Save settings in datastore. Namespace `user-extended-app` key `LOGGER`
- [x] Integrate d2-logger after saving users

### :video_camera: Screenshots/Screen capture

**Settings icon at the bottom of the page**
![image](https://github.com/EyeSeeTea/user-extended-app/assets/461124/7ae6706e-6acb-475e-a896-c803506f2c79)

**D2-Logger Settings Page**
![image](https://github.com/EyeSeeTea/user-extended-app/assets/461124/ddb4b432-f48a-4147-91b3-65083a40469a)

**Events in tracker**
![image](https://github.com/EyeSeeTea/user-extended-app/assets/461124/0c6548f3-1b38-4526-8398-cfb62fde67d9)


### :fire: Testing

- In order to setup d2-logger you will need a tracker program with: 1 attribute of type text to save the username, 1 program stage with 2 dataElements. A dataElement of type FILE and another one of type DATE/DATETIME. (you can see a metadata example in the issue)

- Click in the ⚙️ icon at the bottom of the page
- Select the tracker program you want to use for d2-logger
- Configure the attribute and dataElements.
- Click Save.
- Import some users from a csv file. Then go to the tracker capture app to check the generated logs.

#8694wqxd5
